atlp-rwanda/stackup2-barefoot-backend
# #171380741 Setup API documentation

#### What does this PR do?
Setup API documentation using Swagger
#### Description of Task to be completed?
* Initialize an express server in index.js file
* Create a basic swagger.json spec file
* Install `express`, `swagger-ui-express` library
* Create a sample route at `/api`
#### How should this be manually tested?
1. Open your terminal and run `git clone https://github.com/Stackup-Rwanda/stackup2-barefoot-backend/`
2. run `cd stackup2-barefoot-backend` to go into the root directory of the project
3. run `git checkout -b ch-setup-api-documentation-171380741 origin/ch-setup-api-documentation-171380741` to pull this branch and switch to it
4. run `npm install` to install the dependencies
5. run `npm run start` to start the server
6. Open your browser and navigate to this link: [localhost:3000/public/api-docs](http://localhost:3000/public/api-docs)
7. Click on the highlighted `GET` route, then click on the `Try it out` button then `execute` button and you should see a server response with a status of `200` and a message of `welcome` 
* When this PR is merged, visit the link at step 6 above [here](https://stackup2-bar-ch-setup-a-ouacfx.herokuapp.com/public/api-docs) 
#### Any background context you want to provide?
* You need to install [Node.js](https://nodejs.org/en/) and [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) before testing.
#### What are the relevant pivotal tracker stories?
[#171380741](https://www.pivotaltracker.com/story/show/171380741)
#### Screenshots (if appropriate)
![Barefoot Nomad API Doc 1](https://user-images.githubusercontent.com/45935765/75673982-dc55aa80-5c8c-11ea-8f17-895ef59c33dc.png)
![Barefoot Nomad API Doc 2](https://user-images.githubusercontent.com/45935765/75674108-1757de00-5c8d-11ea-88a0-cbb793c2e067.png)

#### Questions:
